persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/ExpireEmbargoService.java

```diff
@@ -0,0 +1,187 @@
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.TransactionManager;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.metrics.TimerFactory;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.services.OperationsMessageSender;
+import edu.unc.lib.dl.sparql.SparqlQueryService;
+import edu.unc.lib.dl.util.DateTimeUtil;
+import edu.unc.lib.dl.util.JMSMessageUtil;
+import edu.unc.lib.dl.util.SoftwareAgentConstants.SoftwareAgent;
+import io.dropwizard.metrics5.Timer;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+import static edu.unc.lib.dl.rdf.CdrAcl.embargoUntil;
+import static edu.unc.lib.dl.util.DateTimeUtil.formatDateToUTC;
+import static edu.unc.lib.dl.util.DateTimeUtil.parseUTCToDate;
+
+/**
+ * Service that manages embargo expiration
+ *
+ * @author smithjp
+ *
+ */
+@Component
+@EnableScheduling
+public class ExpireEmbargoService {
+
+    private RepositoryObjectFactory repoObjFactory;
+    private RepositoryObjectLoader repoObjLoader;
+    private OperationsMessageSender operationsMessageSender;
+    private TransactionManager txManager;
+    private SparqlQueryService sparqlQueryService;
+
+    private static final Timer timer = TimerFactory.createTimerForClass(ExpireEmbargoService.class);
+
+    private static final Logger log = LoggerFactory.getLogger(ExpireEmbargoService.class);
+
+    public ExpireEmbargoService() {
+    }
+
+    // run service every day 1 minute after midnight
+    @Scheduled(cron = "0 1 0 * * *")
+    public void expireEmbargoes() {
+        log.info("running ExpireEmbargoService");
+        // get list of expired embargoes
+        List<String> resourceList = getEmbargoInfo();
+        Collection<PID> pids = new ArrayList<>();
+        PID currentPid = null;
+
+        if (resourceList.size() > 0) {
+            // remove all expired embargoes
+            for (String rescUri: resourceList) {
+                FedoraTransaction tx = txManager.startTransaction();
+
+                try (Timer.Context context = timer.time()) {
+                    PID pid = PIDs.get(rescUri);
+                    currentPid = pid;
+                    RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+                    Resource resc = repoObj.getResource();
+
+                    // remove embargo
+                    String embargoDate = resc.getProperty(embargoUntil).getString();
+                    repoObjFactory.deleteProperty(repoObj, embargoUntil);
+                    pids.add(pid);
```

Comment:
Might be nice to add a log statement at `debug` or `info` level to say that an embargo was cleared for a particular object in case we have to debug this at some point